nimlangserver.nim

```diff
@@ -240,6 +248,10 @@ proc uriToStash(ls: LanguageServer, uri: string): string =
   else:
     result = ""
 
+proc projectDir(ls: LanguageServer, uri: string): Future[string] {.async.} =
+  ## Returns the directory that a project resides in
+  ls.openFiles[uri].projectFile.await().parentDir()
```

Comment:
I think that we should use rootUri as we do here: https://github.com/yyoncho/langserver/blob/master/nimlangserver.nim#L116 . There is nothing that we can assume about projectFile location.

Reply:
Just consider that we have started the server for `tests/all.nim`. Then any definition will be outside of `/tests/` folder but it will be still fine to do the renaming.

Reply:
Thanks for picking that up!
Can't believe I keep missing this stuff